metricbeat/helper/prometheus/metric.go

```diff
@@ -135,28 +135,33 @@ func (m *commonMetric) GetField() string {
 func (m *commonMetric) GetValue(metric *dto.Metric) interface{} {
 	counter := metric.GetCounter()
 	if counter != nil {
-		return int64(counter.GetValue())
+		if !math.IsNaN(counter.GetValue()) && !math.IsInf(counter.GetValue(), 0) {
+			return int64(counter.GetValue())
+		}
 	}
 
 	gauge := metric.GetGauge()
 	if gauge != nil {
-		return gauge.GetValue()
+		if !math.IsNaN(gauge.GetValue()) && !math.IsInf(gauge.GetValue(), 0) {
+			return gauge.GetValue()
+		}
 	}
 
 	summary := metric.GetSummary()
 	if summary != nil {
 		value := common.MapStr{}
-		value["sum"] = summary.GetSampleSum()
-		value["count"] = summary.GetSampleCount()
+		if !math.IsNaN(summary.GetSampleSum()) && !math.IsInf(summary.GetSampleSum(), 0) {
+			value["sum"] = summary.GetSampleSum()
+			value["count"] = summary.GetSampleCount()
+		}
```

Comment:
Should we also do this same check in the histogram sum?

Reply:
haha I removed it because histogram `bucket` value are `uint64` type instead of `float64` so it will not be NaN or Inf. But instead, it will be `9223372036854775808` 😄 I added the check back into histogram so  we won't get huge numbers like this 👍

Reply:
Well, I said it mainly for the `sum`, that is a `float64`, but I see you added the code for all the values :+1:

Reply:
yeah `sum` is float64 and the other values are uint 😬